packages/tools/src/store/ToolGroupManager/ToolGroup.ts

```diff
@@ -765,7 +773,7 @@ export default class ToolGroup {
       get(this._toolInstances[toolName].configuration, configurationPath) ||
       this._toolInstances[toolName].configuration;
 
-    return structuredClone(_configuration);
+    return utilities.deepClone(_configuration);
```

Comment:
Why is this one deepClone and not structuredClone?

Reply:
Since `structuredClone` cannot copy functions or classes, our custom `deepClone` function uses `structuredClone` for objects while also handling functions and classes.